src/libs/actions/Report.js

```diff
@@ -1543,6 +1464,112 @@ function renameReport(reportID, reportName) {
         .finally(() => Onyx.set(ONYXKEYS.IS_LOADING_RENAME_POLICY_ROOM, false));
 }
 
+/**
+ * @param {Number} reportID
+ * @param {Object} action
+ */
+function viewNewReportAction(reportID, action) {
+    const newMaxSequenceNumber = action.sequenceNumber;
+    const isFromCurrentUser = action.actorAccountID === currentUserAccountID;
+
+    // When handling an action from the current users we can assume that their
+    // last read actionID has been updated in the server but not necessarily reflected
+    // locally so we must first update it and then calculate the unread (which should be 0)
+    if (isFromCurrentUser) {
+        setLocalLastRead(reportID, newMaxSequenceNumber);
+    }
+
+    const updatedReportObject = {
+        // Use updated lastReadSequenceNumber, value may have been modified by setLocalLastRead
+        // Here deletedComments count does not include the new action being added. We can safely assume that newly received action is not deleted.
+        unreadActionCount: newMaxSequenceNumber - (lastReadSequenceNumbers[reportID] || 0) - ReportActions.getDeletedCommentsCount(reportID, lastReadSequenceNumbers[reportID] || 0),
+    };
+
+    Onyx.merge(`${ONYXKEYS.COLLECTION.REPORT}${reportID}`, updatedReportObject);
+
+    // If chat report receives an action with IOU and we have an IOUReportID, update IOU object
+    if (action.actionName === CONST.REPORT.ACTIONS.TYPE.IOU && action.originalMessage.IOUReportID) {
```

Comment:
Similar comment for this logic as for above.

Reply:
I think the answer is the same here. If you miss this update then we just need to have the "reconnection" callbacks account for this data. All the IOU reports should ideally be sent whenever we are "reconnecting".